### PR TITLE
plugins/cilium-cni: add Interface index to the CNI result

### PR DESCRIPTION
**Pull request by mazzy89**
Signed-off-by: Salvatore Mazzarino <dev@mazzarino.cz>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

This adds the corresponding Interface index to the CNI result.

CNI plugins like `tc-redirect-tap` expect this field to be filled.

```release-note
add Interface index to the CNI result
```

### CODE REVIEW

**Comment by mazzy89** [comment]
Slack thread: https://cilium.slack.com/archives/C1MATJ5U5/p1596013346482600

File: plugins/cilium-cni/cilium-cni.go [hunk 1]
```diff
@@ -515,6 +515,11 @@ func cmdAdd(args *skel.CmdArgs) (err error) {
 		Sandbox: args.Netns,
 	})
 
+	// Add to the result the Interface as index of Interfaces
+	for i := range res.Interfaces {
```
Comment (mazzy89) [inline plugins/cilium-cni/cilium-cni.go:519]: AFAIU the binding between `Interfaces` and `IPs` is positional so this should be enough.

**Review by christarazi — APPROVED** [review]

**Comment by christarazi** [comment]
test-me-please

**Comment by mazzy89** [comment]
several tests failed but at a quick glance not clear at the moment if it is due for this change. need to check deeper.

**Comment by christarazi** [comment]
> several tests failed but at a quick glance not clear at the moment if it is due for this change. need to check deeper.

GKE is failing across the board at the moment, so it looks like this PR has passed CI.

**Review by gandro — APPROVED** [review]
Thanks!

**Comment by mazzy89** [comment]
Thank you guys to pushing this out so quickly.

**Pull request merged**